### PR TITLE
Adding PreferredEnclaveType support for Elastic Pools in Az.Sql cmdlets corresponding to 2022-08-01-preview swagger release

### PR DESCRIPTION
**Pull request by vidit-msft**
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by VeryEarly** [comment]
hi @vidit-msft , please record new test cases and existing test cases may have affected by API version upgrade.
For test cases recording please see instruction: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-management-and-platforms/control-plane-bburns/azure-cli-tools-azure-cli-powershell-and-terraform/azure-cli-tools/onboarding/azurepowershell/dev_guidance_sdk#recordingrunning-tests

**Comment by msJinLei** [comment]
@vidit-msft when do you want to release your PR?

**Comment by vidit-msft** [comment]
> @vidit-msft when do you want to release your PR?

@msJinLei We want these changes to be part of the next Az PS release. Is the below information regarding next release accurate or is there a release before this date as well?

image

**Comment by msJinLei** [comment]
```
2023-04-26T22:08:09.7758981Z   [31;1mForEach-Object: [31;1mException calling "NodeModelToMamlModel" with "1" argument(s):
2023-04-26T22:08:09.7760511Z   "D:\a\_work\1\s\artifacts\Debug\Az.Sql\help\Get-AzSqlDatabase.md:253:(563) '  ## Example 7: Get a database configured
2023-04-26T22:08:09.7762527Z   with p...'  Unexpected header name Example 7: Get a database configured with preferred enclave type as VBS"

023-04-26T22:05:57.2948440Z   [xUnit.net 00:00:44.01]     Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ElasticPoolCrudTests.TestElasticPoolGetWithPreferredEnclaveType [FAIL]
2023-04-26T22:05:57.7418158Z     X Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ElasticPoolCrudTests.TestElasticPoolGetWithPreferredEnclaveType [5ms]
2023-04-26T22:05:57.7426617Z EXEC : error Message:  [D:\a\_work\1\s\build.proj]
2023-04-26T22:05:57.7427959Z      System.ArgumentException : Unable to find recorded mock file 'D:\a\_work\1\s\src\Sql\Sql.Test\SessionRecords\Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ElasticPoolCrudTests\TestElasticPoolGetWithPreferredEnclaveType.json'. (Parameter 'callerIdentity')
2023-04-26T22:05:57.7428980Z     Stack Trace:
2023-04-26T22:05:57.7429820Z        at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity, HttpRecorderMode mode) in D:\a\_work\1\s\tools\TestFx\DelegatingHandlers\HttpMockServer.cs:line 83
2023-04-26T22:05:57.7431105Z      at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity) in D:\a\_work\1\s\tools\TestFx\DelegatingHandlers\HttpMockServer.cs:line 55
2023-04-26T22:05:57.7432101Z      at Microsoft.Rest.ClientRuntime.Azure.TestFramework.MockContext.Start(String className, String methodName) in D:\a\_work\1\s\tools\TestFx\Mocks\MockContext.cs:line 67
2023-04-26T22:05:57.7433067Z      at Microsoft.Azure.Commands.TestFx.TestManager.RunTestScriptCore(Action setUp, Action`1 contextAction, Action tearDown, String[] scripts) in D:\a\_work\1\s\tools\TestFx\TestManager.cs:line 249
2023-04-26T22:05:57.7434023Z      at Microsoft.Azure.Commands.TestFx.TestManager.RunTestScript(String[] scripts) in D:\a\_work\1\s\tools\TestFx\TestManager.cs:line 221
2023-04-26T22:05:57.7435013Z      at Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ElasticPoolCrudTests.TestElasticPoolGetWithPreferredEnclaveType() in D:\a\_work\1\s\src\Sql\Sql.Test\ScenarioTests\ElasticPoolCrudTests.cs:line 167
2023-04-26T22:06:04.2415906Z   [xUnit.net 00:00:50.96]     Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ElasticPoolCrudTests.TestElasticPoolCreateWithPreferredEnclaveType [FAIL]
2023-04-26T22:06:05.7537432Z     X Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ElasticPoolCrudTests.TestElasticPoolCreateWithPreferredEnclaveType [1ms]
2023-04-26T22:06:05.7538007Z EXEC : error Message:  [D:\a\_work\1\s\build.proj]
2023-04-26T22:06:05.7538675Z      System.ArgumentException : Unable to find recorded mock file 'D:\a\_work\1\s\src\Sql\Sql.Test\SessionRecords\Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ElasticPoolCrudTests\TestElasticPoolCreateWithPreferredEnclaveType.json'. (Parameter 'callerIdentity')
2023-04-26T22:06:05.7539304Z     Stack Trace:
2023-04-26T22:06:05.7539874Z        at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity, HttpRecorderMode mode) in D:\a\_work\1\s\tools\TestFx\DelegatingHandlers\HttpMockServer.cs:line 83
2023-04-26T22:06:05.7540714Z      at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity) in D:\a\_work\1\s\tools\TestFx\DelegatingHandlers\HttpMockServer.cs:line 55
2023-04-26T22:06:05.7541438Z      at Microsoft.Rest.ClientRuntime.Azure.TestFramework.MockContext.Start(String className, String methodName) in D:\a\_work\1\s\tools\TestFx\Mocks\MockContext.cs:line 67
2023-04-26T22:06:05.7542110Z      at Microsoft.Azure.Commands.TestFx.TestManager.RunTestScriptCore(Action setUp, Action`1 contextAction, Action tearDown, String[] scripts) in D:\a\_work\1\s\tools\TestFx\TestManager.cs:line 249
2023-04-26T22:06:05.7543044Z      at Microsoft.Azure.Commands.TestFx.TestManager.RunTestScript(String[] scripts) in D:\a\_work\1\s\tools\TestFx\TestManager.cs:line 221
2023-04-26T22:06:05.7543724Z      at Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ElasticPoolCrudTests.TestElasticPoolCreateWithPreferredEnclaveType() in D:\a\_work\1\s\src\Sql\Sql.Test\ScenarioTests\ElasticPoolCrudTests.cs:line 83
2023-04-26T22:06:07.3701584Z   [xUnit.net 00:00:54.08]     Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ElasticPoolCrudTests.TestElasticPoolUpdateWithPreferredEnclaveType [FAIL]
2023-04-26T22:06:08.8277534Z     X Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ElasticPoolCrudTests.TestElasticPoolUpdateWithPreferredEnclaveType [1ms]
2023-04-26T22:06:08.8277991Z EXEC : error Message:  [D:\a\_work\1\s\build.proj]
2023-04-26T22:06:08.8278535Z      System.ArgumentException : Unable to find recorded mock file 'D:\a\_work\1\s\src\Sql\Sql.Test\SessionRecords\Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ElasticPoolCrudTests\TestElasticPoolUpdateWithPreferredEnclaveType.json'. (Parameter 'callerIdentity')
2023-04-26T22:06:08.8279068Z     Stack Trace:
2023-04-26T22:06:08.8279565Z        at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity, HttpRecorderMode mode) in D:\a\_work\1\s\tools\TestFx\DelegatingHandlers\HttpMockServer.cs:line 83
2023-04-26T22:06:08.8282483Z      at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity) in D:\a\_work\1\s\tools\TestFx\DelegatingHandlers\HttpMockServer.cs:line 55
2023-04-26T22:06:08.8288787Z      at Microsoft.Rest.ClientRuntime.Azure.TestFramework.MockContext.Start(String className, String methodName) in D:\a\_work\1\s\tools\TestFx\Mocks\MockContext.cs:line 67
2023-04-26T22:06:08.8289544Z      at Microsoft.Azure.Commands.TestFx.TestManager.RunTestScriptCore(Action setUp, Action`1 contextAction, Action tearDown, String[] scripts) in D:\a\_work\1\s\tools\TestFx\TestManager.cs:line 249
2023-04-26T22:06:08.8290117Z      at Microsoft.Azure.Commands.TestFx.TestManager.RunTestScript(String[] scripts) in D:\a\_work\1\s\tools\TestFx\TestManager.cs:line 221
2023-04-26T22:06:08.8290649Z      at Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ElasticPoolCrudTests.TestElasticPoolUpdateWithPreferredEnclaveType() in D:\a\_work\1\s\src\Sql\Sql.Test\ScenarioTests\ElasticPoolCrudTests.cs:line 132
2023-04-26T22:07:46.8500897Z   [xUnit.net 00:02:32.58]     Microsoft.Azure.Commands.Sql.Test.ScenarioTests.DatabaseCrudTests.TestDatabaseUpdateWithPreferredEnclaveType [FAIL]
2023-04-26T22:07:47.7271426Z     X Microsoft.Azure.Commands.Sql.Test.ScenarioTests.DatabaseCrudTests.TestDatabaseUpdateWithPreferredEnclaveType [1ms]
2023-04-26T22:07:47.7272019Z EXEC : error Message:  [D:\a\_work\1\s\build.proj]
2023-04-26T22:07:47.7272913Z      System.ArgumentException : Unable to find recorded mock file 'D:\a\_work\1\s\src\Sql\Sql.Test\SessionRecords\Microsoft.Azure.Commands.Sql.Test.ScenarioTests.DatabaseCrudTests\TestDatabaseUpdateWithPreferredEnclaveType.json'. (Parameter 'callerIdentity')
2023-04-26T22:07:47.7273521Z     Stack Trace:
2023-04-26T22:07:47.7274100Z        at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity, HttpRecorderMode mode) in D:\a\_work\1\s\tools\TestFx\DelegatingHandlers\HttpMockServer.cs:line 83
2023-04-26T22:07:47.7274930Z      at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity) in D:\a\_work\1\s\tools\TestFx\DelegatingHandlers\HttpMockServer.cs:line 55
2023-04-26T22:07:47.7275648Z      at Microsoft.Rest.ClientRuntime.Azure.TestFramework.MockContext.Start(String className, String methodName) in D:\a\_work\1\s\tools\TestFx\Mocks\MockContext.cs:line 67
2023-04-26T22:07:47.7276376Z      at Microsoft.Azure.Commands.TestFx.TestManager.RunTestScriptCore(Action setUp, Action`1 contextAction, Action tearDown, String[] scripts) in D:\a\_work\1\s\tools\TestFx\TestManager.cs:line 249
2023-04-26T22:07:47.7277062Z      at Microsoft.Azure.Commands.TestFx.TestManager.RunTestScript(String[] scripts) in D:\a\_work\1\s\tools\TestFx\TestManager.cs:line 221
2023-04-26T22:07:47.7277785Z      at Microsoft.Azure.Commands.Sql.Test.ScenarioTests.DatabaseCrudTests.TestDatabaseUpdateWithPreferredEnclaveType() in D:\a\_work\1\s\src\Sql\Sql.Test\ScenarioTests\DatabaseCrudTests.cs:line 123
```

File: src/Sql/Sql/ChangeLog.md [hunk 1]
```diff
@@ -27,6 +27,7 @@
 ## Version 4.5.0
 * Added a new cmdlet to refresh external governance status
     - 'Invoke-AzSqlServerExternalGovernanceStatusRefresh'
+* Added `PreferredEnclaveType` parameter to `NewAzureSqlElasticPool`, `Get-AzSqlElasticPool` and `SetAzureSqlElasticPool` cmdlet
```
Comment (msJinLei) [inline src/Sql/Sql/ChangeLog.md:30]: Is `PreferredEnclaveType added to `Get-AzSqlElasticPool`? Not find in codes

Comment (msJinLei) [reply src/Sql/Sql/ChangeLog.md]: > oh, so doing a get call on elastic pool with `Get-AzSqlElasticPool` now gives out the new parameter `PreferredEnclaveType` as well, but there are no changes made to the get cmdlet as such to support this change.

What do you mean by “Added `PreferredEnclaveType` parameter `Get-AzSqlElasticPool` cmdlet”?

Comment (vidit-msft) [reply src/Sql/Sql/ChangeLog.md]: updated the ChangeLog.md

**Review by msJinLei — APPROVED** [review]

**Review by msJinLei — APPROVED** [review]

**Pull request merged**